code/modules/surgery/implant_removal.dm

```diff
@@ -1,81 +1,199 @@
+
+//extract implant
+/datum/surgery_step/destroy_extract_implant
+	name = "destroy/extract implant (drill/hemostat)"
+	repeatable = TRUE
+
+	// Destruction tools
+	implements = list(
+		TOOL_DRILL = 100,
+		/obj/item/screwdriver/power = 80,
+		/obj/item/pickaxe/drill = 60,
+		TOOL_SCREWDRIVER = 25,
+	)
+	var/implements_extract = list(
+		TOOL_HEMOSTAT = 100,
+		TOOL_CROWBAR = 55,
+		/obj/item/kitchen/fork = 35,
+		/obj/item/kitchen/spoon = 20,
+	)
+	time = 6.4 SECONDS
+	success_sound = 'sound/surgery/hemostat1.ogg'
+	var/obj/item/implant/target_implant
+	var/obj/item/implantcase/target_case
+
+/datum/surgery_step/destroy_extract_implant/New()
+	..()
+	implements = implements + implements_extract
 
+/datum/surgery_step/destroy_extract_implant/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/implant_removal/surgery)
+	if(!length(surgery.discovered_implants))
+		to_chat(user, span_warning("No implants in [target]'s [parse_zone(target_zone)] have been discovered, or they've all been destroyed/extracted!"))
+		return SURGERY_STEP_FAIL
+
+	for(var/obj/item/discovered in surgery.discovered_implants)
+		if(!(discovered in target.implants))
+			surgery.discovered_implants -= discovered
+
+	if(implement_type in implements_extract)
 		var/obj/item/implantcase/case
 		for(var/obj/item/implantcase/implant_case in user.held_items)
-			case = implant_case
-			break
+			if(!implant_case.imp)
+				case = implant_case
+				break
 		if(!case)
-			case = locate(/obj/item/implantcase) in get_turf(target)
-		if(case && !case.imp)
-			case.imp = implant
-			implant.forceMove(case)
-			case.update_appearance()
+			for(var/obj/item/implantcase/implant_case in range(1, user))
+				if(!implant_case.imp)
+					case = implant_case
+					break
+		if(!case)
+			to_chat(user, span_warning("You have no implant case in-hand or nearby to store extracted implants from [target]. Did you mean to destroy the implant?"))
+			return SURGERY_STEP_FAIL
+
+		var/chosen_implant = tgui_input_list(user, "Extract which implant?", "Surgery", sort_list(surgery.discovered_implants))
+		if(isnull(chosen_implant) || !(chosen_implant in target.implants))
+			return SURGERY_STEP_FAIL
+		if(user && target && user.Adjacent(target) && user.get_active_held_item() == tool && case && user.Adjacent(case) && !case.imp)
+			target_implant = chosen_implant
+			target_case = case
+			display_results(
+				user,
+				target,
+				span_notice("You begin to extract [chosen_implant] from [target]'s [parse_zone(target_zone)]..."),
+				span_notice("[user] begins to extract [chosen_implant] from [target]'s [parse_zone(target_zone)]."),
+				span_notice("[user] begins to extract something from [target]'s [parse_zone(target_zone)]."),
+			)
+			// I'm sorry, but you cannot tell what type of implants is being removed from sensation alone.
+			display_pain(target, "You can feel an implant being removed from your [parse_zone(target_zone)]!")
+		else
+			return SURGERY_STEP_FAIL
+
+	else
```

Comment:
Does this need to be an else, or can you just ensure that the previous `if` always returns eventually?